packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -344,6 +355,17 @@ export function recmaJsxRewrite(options = {}) {
               componentsInit = {type: 'Identifier', name: '_components'}
             }
 
+            for (const [
+              invalidComponentName,
+              variable
+            ] of invalidComponentNameToVariable) {
```

Comment:
I’d prefer for this to use a bit of a simpler loop. While, a `for .. in`, like much of the other code.
Or at least I find the destructuring a confusing to read, so assigning it to on variable, and use that?

Reply:
I understand, and I think the variables could have better names! However, I don't know an easier way to iterate over a JS `Map` with type safety (note this is a `Map` and not an object). I tried this refactor using a `for ... in` over the keys as you suggested:

```js
for (const literalValue in invalidComponentNameToVariable) {
  const identifierName = invalidComponentNameToVariable.get(literalValue)
  declarations.push({
    type: 'VariableDeclarator',
    id: {type: 'Identifier', name: identifierName},
    init: {type: 'Literal', value: literalValue}
  })
}
```
...but `identifierName` is of type `string | undefined`, since we can't guarantee a result from a `.get` call. This throws a type error on `name`!

Respectfully, I think this `for ... of` solution is the most readable way to iterate over a `Map` in this case. What do you think about keeping the `for ... of` with this variable naming?

```js
for (const [identifierName, literalValue] of invalidComponentNameToVariable) {
  declarations.push({
    type: 'VariableDeclarator',
    id: {type: 'Identifier', name: identifierName},
    init: {type: 'Literal', value: literalValue}
  })
}
```

Reply:
Alright, sure!

You could use `name, value` as identifiers, and then have them be shortcuts in the declaration. It’s a small loop body, those names are very clear as well?